Dynatrace/dynatrace-operator
# Fix failing initContainer incase of classicFullStack and metadataEnrichment combination

Ticket: https://dt-rnd.atlassian.net/browse/K8S-10259

## Description

Since we want to combine classicFullStack and metaDataEnrichment, combining those two will cause an error -> the agent mounts over /var/lib/dynatrace/enrichment, causing the initContainer to fail with createFile errors (lack of permission) on the hostFileSystem.
We should instead of directly creating the files there, create them in a temp location and mount the with subPath (only files not the whole folder) 
Afterwards the agent will make modifications to not overwrite our files

## How can this be tested?

Apply a Dynakube with metadata enrichment only and check if all files (.json and .properties are created)
Apply another Dynakube afterwards with classicFullStack and metadataEnrichment to check if the containers start successfully (note: the files will still be gone, but the agent has a ticket to resolve this)
